OpenCover.UI.TestDiscoverer/NUnit/NUnitDiscoverer.cs

```diff
@@ -111,7 +135,11 @@ private TestMethod[] DiscoverTestsInClass(TypeDefinition type, TestClass @class)
 			foreach (var method in type.Methods)
 			{
 				bool isTestMethod = false;
-				var trait = new List<string>();
+                var traits = new List<string>();
+                if (@class.Traits != null && @class.Traits.Length > 0)
+                {
+                    traits.AddRange(@class.Traits);
+                }
```

Comment:
You could replace this block like this (in combination with not setting Traits=null, see other comment) :
`var traits = @class.Traits.ToList()`
This takes a copy of the array without having to copy it manually